### ISSUE TITLE
Graph-based autoencoder

### ISSUE THREAD

**Issue by Kai-Zhu-2001** [luigibonati/mlcolvar#274]
We should implement a graph-based autoencoder for learning low-dimensional collective variables from molecular configurations. The encoder should take a molecular graph as input,  apply graph pooling to obtain a global representation, and then use an MLP readout to map this representation into latent variables. These latent variables will be used as CVs for biasing and interpretation. During training, a decoder maps the latent variables back to the pooled graph-level representation produced by the GNN encoder, rather than reconstructing the full molecular graph.  After training, the decoder is discarded, and only the GNN encoder is used for CV evaluation and enhanced sampling.

**Comment by Kai-Zhu-2001** [comment]
In addition, we can start from an already trained GNN-based model, extract its learned feature representation, and then train an MLP autoencoder to reconstruct these features through a low-dimensional bottleneck. In this way, the bottleneck variables can be used as collective variables while preserving the structural information captured by the original GNN. This strategy is also general and can be extended to any MLCV model by applying the autoencoder to the learned feature representation rather than to the raw input.

**Comment by Kai-Zhu-2001** [comment]
### Update

With the pretrained-representation framework introduced in #283, this functionality can now be implemented without adding a dedicated graph-autoencoder class or modifying the existing `AutoEncoderCV`.

The proposed workflow is:

```text
molecular graph
      ↓
pretrained graph representation
      ↓
frozen pooled features
      ↓
AutoEncoderCV
      ↓
latent CV
```

During training, the graph representation is kept frozen and can optionally be precomputed and cached. The existing `AutoEncoderCV` is then trained directly on these fixed graph-level features using its standard reconstruction loss.

For CV evaluation and enhanced sampling, the pretrained representation and the trained `AutoEncoderCV` can be composed through `RepresentationModel`:

```text
graph
  ↓
frozen pretrained model
  ↓
graph-level representation
  ↓
AutoEncoderCV.forward()
  ↓
latent CV
```

Since `AutoEncoderCV.forward()` already evaluates only the encoder branch, the decoder is used only during training and does not participate in CV evaluation.

This approach has a few advantages:

* no graph-specific changes are required in `AutoEncoderCV`;
* the reconstruction target is fixed, avoiding collapse associated with jointly learning and reconstructing the same GNN representation;
* the same workflow can be used with different pretrained graph/atomistic representations;
* cached representations can be used for efficient AutoEncoder training;
* gradients with respect to atomic coordinates are preserved during deployment through the representation framework.

The remaining work for this issue could therefore be mainly:

* add an example/tutorial showing pretrained graph representation → `AutoEncoderCV`;
* add a test comparing CVs obtained from cached representations with those obtained from the composed raw-input model;
* test TorchScript/Metatomic export for the composed model.

If an end-to-end trainable GNN autoencoder is still desired, where the GNN representation itself is optimized jointly with the reconstruction objective, I would treat that as a separate extension because it requires an additional strategy to prevent representation collapse.
